tasks/main.yaml

```diff
@@ -42,6 +42,21 @@
     regexp: '^GRUB_CMDLINE_LINUX="'
   when: >
     ansible_os_family == "RedHat" and
-    ansible_distribution_major_version | int >= 7
+    ansible_distribution_major_version | int >= 7 and
+    grub_cmdline_add_args | length > 0
+  notify:
+    - Update Grub2 config file
+
+- name: Remove arguments from Grub default config for RedHat >= 7
+  lineinfile:
+    dest: "{{ grub_cmdline_default }}"
+    regexp: '^(.*){{ item }}(.*)$'
```

Comment:
Not sure what you mean by GRUB_CMDLINE_LINUX_DEFAULT. I think you meant that it should only check for the GRUB_CMDLINE_LINUX line in /etc/default/grub. Can you clarify?